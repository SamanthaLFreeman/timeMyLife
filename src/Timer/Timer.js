import React from 'react';

const Timer = () => {
  return (
    <section className='ActivityForm'>
      <h3>Name</h3>
      <p>Timer</p>
      <button>Start</button>
      <button>Cancel Activity</button>
    </section>
  )
}

export default Timer;