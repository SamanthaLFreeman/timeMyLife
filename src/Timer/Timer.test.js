import React from 'react';
import Timer from './Timer';
import { shallow } from 'enzyme';

describe('Timer', () => {
  let wrapper;
  const currentActivityMock = {
    name: 'Activity Name'
  }
  const minsMock = '1'
  const secsMock = '30'
  const startTimerMock = jest.fn()
  const removeCurrentActivityMock = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <Timer 
        removeCurrentActivity={removeCurrentActivityMock}
        currentActivity={currentActivityMock}
        mins={minsMock}
        secs={secsMock}
        startTimer={startTimerMock} />)
  })

  it('should match the snapshot with the correct data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})