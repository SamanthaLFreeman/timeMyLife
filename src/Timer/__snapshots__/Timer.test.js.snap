// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer should match the snapshot with the correct data passed through 1`] = `
<section
  className="ActivityForm"
>
  <h3>
    Activity Name
  </h3>
  <p>
    Time Remaining: 
    1
    :
    30
  </p>
  <button
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    Start
  </button>
  <button
    onClick={[MockFunction]}
  >
    Cancel Activity
  </button>
</section>
`;
